src/main/java/sirius/biz/storage/layer3/Transfer.java

```diff
@@ -331,26 +331,29 @@ private void transfer(boolean delete) {
                 destination.createAsDirectory();
             }
 
-            transferDirectory(source, destination.findChild(source.name()), delete);
+            transferDirectory(source, destination.findChild(source.name()), delete, delete);
         }
     }
 
-    private void transferDirectory(VirtualFile sourceDirectory, VirtualFile destinationDirectory, boolean delete) {
+    private void transferDirectory(VirtualFile sourceDirectory,
+                                   VirtualFile destinationDirectory,
+                                   boolean deleteContent,
+                                   boolean deleteSourceDirectory) {
         if (!destinationDirectory.isDirectory()) {
             destinationDirectory.createAsDirectory();
         }
 
         sourceDirectory.allChildren().stream().forEach(child -> {
             if (child.isFile()) {
-                transferFileTo(child, destinationDirectory.findChild(child.name()), delete);
+                transferFileTo(child, destinationDirectory.findChild(child.name()), deleteContent);
             } else {
-                transferDirectory(child, destinationDirectory.findChild(child.name()), delete);
+                transferDirectory(child, destinationDirectory.findChild(child.name()), deleteContent, deleteContent);
```

Comment:
Even if thats correct this looks like an error, so maybe explain why the same param is passed twice here :)